source/iface/usm_allocations.rst

```diff
@@ -31,99 +59,48 @@ declaring standard library containers (e.g. vector).
 ``Alignment``  Alignment of the allocation
 =============  ===
 
-.. rubric:: Example
-
-.. literalinclude:: /examples/std-vector.cpp
-   :lines: 5-
-   :linenos:
-
-.. rubric:: Member types
-
-==============  ===
-``value_type``
-==============  ===
-
-.. seealso:: `SYCL Specification <https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_c_allocator_interface>`__
+There is no specialization for ``usm::alloc::device`` because an ``Allocator``
+is required to allocate memory that is accessible on the host.
 
 (constructors)
 ==============
 
 ::
 
-  usm_allocator(const sycl::context &ctxt, const sycl::device &dev) noexcept;
-  usm_allocator(const sycl::queue &q) noexcept;
-  usm_allocator(const sycl::usm_allocator &other) noexcept;
-  template <class U>
-  usm_allocator(sycl::usm_allocator<U, AllocKind, Alignment> const &) noexcept;
-
-``allocate``
-============
-
-::
-
-  T *allocate(size_t Size);
-
-Allocates memory
-
-``deallocate``
-==============
-
-::
-
-  void deallocate(T *Ptr, size_t size);
-
-Deallocates memory
-
-``construct``
-=============
-
-::
-
-  template <
-      sycl::usm::alloc AllocT = AllocKind,
-      typename std::enable_if<AllocT != sycl::usm::alloc::device, int>::type = 0,
-      class U, class... ArgTs>
-  void construct(U *Ptr, ArgTs &&... Args);
-  template <
-      sycl::usm::alloc AllocT = AllocKind,
-      typename std::enable_if<AllocT == sycl::usm::alloc::device, int>::type = 0,
-      class U, class... ArgTs>
-  void construct(U *Ptr, ArgTs &&... Args);
+  usm_allocator(const context& syclContext, const device& syclDevice,
+              const property_list& propList = {});
 
-Constructs an object on memory pointed by ``Ptr``.
+  usm_allocator(const queue& syclQueue, const property_list& propList = {});
 
-``destroy``
-===========
+Constructs a ``usm_allocator`` instance that allocates USM for the provided
+context and device presented with first constructor.
+The second constructor present simplified form where ``syclQueue`` provides
+the ``device`` and ``context``.
 
-::
+.. rubric:: Exceptions
 
-  template <
-      sycl::usm::alloc AllocT = AllocKind,
-      typename std::enable_if<AllocT != sycl::usm::alloc::device, int>::type = 0>
-  void destroy(T *Ptr);
+``errc::invalid``
+  If ``syclContext`` does not encapsulate the SYCL ``device`` returned
+  by ``deviceSelector``.
 
-  /// Throws an error when trying to destroy a device allocation
-  /// on the host
-  template <
-      sycl::usm::alloc AllocT = AllocKind,
-      typename std::enable_if<AllocT == sycl::usm::alloc::device, int>::type = 0>
-  void destroy(T *Ptr);
+``errc::feature_not_supported``
+  If ``AllocKind`` is ``usm::alloc::host``, the first constructor throws a synchronous
```

Comment:
This exception (and the following one) apply to both constructors.

Reply:
Fixed with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/81/commits/b9f594cc02c56186f83a5e99463c1a67fe28013c